vpr/src/base/place_and_route.cpp

```diff
@@ -353,16 +353,17 @@ int binary_search_place_and_route(const t_placer_opts& placer_opts_ref,
                     device_ctx.num_arch_switches,
                     det_routing_arch,
                     segment_inf,
-                    router_opts.base_cost_type,
-                    router_opts.trim_empty_channels,
-                    router_opts.trim_obs_channels,
-                    router_opts.clock_modeling,
+                    router_opts,
                     arch->Directs, arch->num_directs,
-                    &warnings,
-                    router_opts.read_rr_edge_metadata,
-                    router_opts.do_check_rr_graph);
+                    &warnings);
 
     init_draw_coords(final);
+
+    /* Allocate and load additional rr_graph information needed only by the router. */
+    alloc_and_load_rr_node_route_structs();
```

Comment:
It seems weird this got added here, but nothing was removed.  I believe you added this because it is required, but can you explain why a call to these methods didn't get removed?

Reply:
A different way of asking this is "why did this work before?"

Reply:
I believe this worked before because the graph would be rebuilt with a graph similar enough on each iteration that it wasn't noticed before. With `random_shuffle`, it gets shuffled on each iteration.

Reply:
This discussion did help me resolve a bug where the graph isn't restored correctly to output the best route, because the PRNG seed was not fixed for `random_shuffle`. Thanks!